PWG/EMCAL/EMCALtasks/AliEmcalMCTrackSelector.cxx

```diff
@@ -126,16 +131,16 @@ void AliEmcalMCTrackSelector::UserExec(Option_t *)
         fEvent->AddObject(fParticlesMap);
       }
 
-      if (!fIsESD) {
-        fParticlesIn = static_cast<TClonesArray*>(InputEvent()->FindListObject(AliAODMCParticle::StdBranchName()));
+      if (!fIsESD || (fIsESD && fIsContainerSpecified)) {
+        fParticlesIn = static_cast<TClonesArray*>(InputEvent()->FindListObject(fTrackContainerName));
```

Comment:
In case of `usedefault` a handling translating this into the real default container name is missing here.